DependencyInjectionContainer/DIContainerBuilder.cs

```diff
@@ -0,0 +1,106 @@
+﻿using DependencyInjectionContainer.Attributes;
+using DependencyInjectionContainer.Enums;
+using DependencyInjectionContainer.Exceptions;
+using System.Reflection;
+
+namespace DependencyInjectionContainer
+{
+    public sealed class DIContainerBuilder
+    {
+        private List<Service> services;
+        private DIContainer parentContainer;
+        private bool isBuild;
+
+        public DIContainerBuilder()
+        {
```

Comment:
You can use class initializer and ditch this constructor
https://stackoverflow.com/questions/24551/initialize-class-fields-in-constructor-or-at-declaration